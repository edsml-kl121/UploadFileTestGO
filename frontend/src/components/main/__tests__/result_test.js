import {render, screen, cleanup} from "@testing-library/react"
import Result from "../helpercomponent/result.component.js";
import renderer from 'react-test-renderer'

afterEach(() => {
  cleanup()
})


test('matches snapshot', () => {
  const tree = renderer.create(<Result props ={[1,2]}/>)
  expect(tree).toMatchSnapshot();
})


test('should have the text, up', () => {
  render(<Result props ={[1,2]}/>)
  const resultElement = screen.getByTestId('up-1')
  expect(resultElement).toBeInTheDocument();
  expect(resultElement).toHaveTextContent('UP');
})

test('should have the right value when passed in props for up', () => {
  render(<Result props ={[1,{up: 1, down: 2}]}/>)
  const resultElement = screen.getByTestId('up-2')
  expect(resultElement).toBeInTheDocument();
  expect(resultElement).toHaveTextContent(1);
})


test('should have the text, down', () => {
  render(<Result props ={[1,2]}/>)
  const resultElement = screen.getByTestId('down-1')
  expect(resultElement).toBeInTheDocument();
  expect(resultElement).toHaveTextContent('DOWN');
})

test('should have the right value when passed in props for down', () => {
  render(<Result props ={[1,{up: 1, down: 2}]}/>)
  const resultElement = screen.getByTestId('down-2')
  expect(resultElement).toBeInTheDocument();
  expect(resultElement).toHaveTextContent(2);
})

test('should have the right title when passed in props status', () => {
  render(<Result props ={[1,{up: 1, down: 2, Total: 3}]}/>)
  const resultElement = screen.getByTestId('resulttitle-1')
  expect(resultElement).toBeInTheDocument();
  expect(resultElement).toHaveTextContent('Total 3 Websites');
})

test('should have the right seconds when passed in props elapsed', () => {
  render(<Result props ={[1,{up: 1, down: 2, Total: 3, Elapsed:40}]}/>)
  const resultElement = screen.getByTestId('second-1')
  expect(resultElement).toBeInTheDocument();
  expect(resultElement).toHaveTextContent('40 second');
})

test('should have the right minute when passed in props elapsed', () => {
  render(<Result props ={[1,{up: 1, down: 2, Total: 3, Elapsed:97}]}/>)
  const resultElement = screen.getByTestId('minute-1')
  expect(resultElement).toBeInTheDocument();
  expect(resultElement).toHaveTextContent('1 minute 37 second');
})