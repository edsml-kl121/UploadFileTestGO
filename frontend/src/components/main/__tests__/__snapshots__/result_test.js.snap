// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="result mb-5"
  >
    <div
      style={
        Object {
          "paddingBottom": "15px",
        }
      }
    >
      <p
        data-testid="resulttitle-1"
        style={
          Object {
            "fontSize": "36px",
            "textAlign": "left",
          }
        }
      >
        Total 
         Websites
      </p>
      <p
        data-testid="minute-1"
        style={
          Object {
            "fontSize": "24px",
            "textAlign": "left",
          }
        }
      >
        (Used 
        NaN
         minute 
        NaN
         second)
      </p>
    </div>
    <div
      className="row"
    >
      <div
        className="resultup col"
      >
        <p
          data-testid="up-1"
          style={
            Object {
              "fontSize": "24px",
              "marginBottom": "0px",
              "textAlign": "left",
            }
          }
        >
           UP 
        </p>
        <span
          className="numbers"
          data-testid="up-2"
        />
      </div>
      <div
        className="resultdown col"
      >
        <p
          data-testid="down-1"
          style={
            Object {
              "fontSize": "24px",
              "marginBottom": "0px",
              "textAlign": "left",
            }
          }
        >
           DOWN 
        </p>
        <span
          className="numbers"
          data-testid="down-2"
        />
      </div>
    </div>
  </div>
</div>
`;
